UnityProject/Assets/Scripts/Objects/Closets/AbandonedCrates.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections.Generic;
+using MiniGames;
+using Mirror;
+using UnityEngine;
+
+namespace Objects.Closets
+{
+	public class AbandonedCrates : ClosetControl, IMiniGame
+	{
+
+		[SerializeField] private MiniGameResultTracker miniGameTracker;
+		[SerializeField] private List<MiniGameModule> miniGameModules;
+		[SyncVar] private int currentMiniGameIndex = -1;
+
+		public override void OnSpawnServer(SpawnInfo info)
+		{
+			base.OnSpawnServer(info);
+			SetupMiniGames();
+		}
+
+		private void SetupMiniGames()
+		{
+			if(miniGameTracker == null) return;
+			if (miniGameModules.Count == 0)
+			{
+				Logger.LogError("[ClosetControl/MiniGames] - Found MiniGame tracker but no minigames found!");
+				return;
+			}
+			currentMiniGameIndex = Random.Range(0, miniGameModules.Count - 1);
+			miniGameModules[currentMiniGameIndex].Setup(miniGameTracker, gameObject);
+			SetLock(Lock.Locked);
+		}
+
+		protected override void InteractionChecks(PositionalHandApply interaction)
```

Comment:
I feel like it would make more sense to override `TryToggleLock`? I'd prefer an event approach, but overriding a more specific method is also fine.

Reply:
Why would you "Try" to do something that should be a forced state?
This feels like it's going to result in bugs, so I'm not taking risks.